hdfs-commons/src/main/java/org/apache/mesos/hdfs/config/HdfsFrameworkConfig.java

```diff
@@ -61,6 +66,25 @@ public HdfsFrameworkConfig() {
     setConf(configuration);
   }
 
+  public void writeXml() {
```

Comment:
Let's separate this method into a separate class. Also, we need to think about concurrency here. What if there are multiple requests coming in at once. We need a form of locking.